tf2onnx/optimizer/transpose_optimizer.py

```diff
@@ -306,6 +307,15 @@ def _create_transpose_pairs_after_node(self, node):
             self._g.replace_input(consumer, node.output[0], nhwc_node.output[0])
 
     def _create_transpose_pairs_before_node(self, node):
+        def shape_after_expand(ori_shape):
+            # according to broadcasting rule to expand shape to 4D while not tile the tensor here
+            # still count on the broadcasting op to tile the tensor
+            utils.make_sure(ori_shape.count(-1) <= 1, "shape can contain one -1 at most")
+            ori_rank = len(ori_shape)
+            utils.make_sure(ori_rank <= 4, "ONNX only supports 4D data")

```

Comment:
this error message does not describe the problem

Reply:
this checker seems useless here since insert transpose means it's not larger than 4D, will remove.